src/main/java/io/github/thebusybiscuit/slimefun4/api/events/AsyncReactorProcessCompleteEvent.java

```diff
@@ -0,0 +1,77 @@
+package io.github.thebusybiscuit.slimefun4.api.events;
+
+import javax.annotation.Nonnull;
+import javax.annotation.ParametersAreNonnullByDefault;
+
+import org.bukkit.Location;
+import org.bukkit.event.Event;
+import org.bukkit.event.HandlerList;
+
+import io.github.thebusybiscuit.slimefun4.implementation.items.electric.reactors.Reactor;
+import me.mrCookieSlime.Slimefun.Objects.SlimefunItem.abstractItems.MachineFuel;
+import me.mrCookieSlime.Slimefun.Objects.SlimefunItem.SlimefunItem;
+
+/**
+ * This {@link Event} is fired whenever a {@link Reactor} has completed its process.
+ *
+ * @author poma123
+ *
+ */
+public class AsyncReactorProcessCompleteEvent extends Event {
```

Comment:
This should probably extend AsyncMachineProcessCompleteEvent

Reply:
But Reactor doesn't extend AContainer. Also, Reactor has MachineFuel, AContainer has MachineRecipe.

Reply:
I'm just saying all these 3 events are very similar, and doing a `event instanceof MachineEvent` or sth. will probably prove usedul to some people.

Reply:
Hmm. Good idea though.